fe/fe-core/src/main/java/com/starrocks/sql/optimizer/task/RewriteTreeTask.java

```diff
@@ -101,6 +100,10 @@ private boolean match(Pattern pattern, OptExpression root) {
             return false;
         }
 
+        if (pattern.children().size() > 0 && pattern.children().size() != root.getInputs().size() &&
+                pattern.children().stream().noneMatch(Pattern::isPatternMultiLeaf)) {
+            return false;
```

Comment:
Is this just for short circuit?

Reply:
for strict check. pattern leaf could match multiple-children before this pr